plugins/woo-ai/src/product-name/product-name-suggestions.tsx

```diff
@@ -56,8 +53,35 @@ export const ProductNameSuggestions = () => {
 	const [ suggestions, setSuggestions ] = useState< ProductDataSuggestion[] >(
 		[]
 	);
-	const { fetchSuggestions } = useProductDataSuggestions();
 	const { updateProductSlug } = useProductSlug();
+	const { requestCompletion } = useCompletion( {
+		onStreamError: ( error ) => {
+			// eslint-disable-next-line no-console
+			console.debug( 'Streaming error encountered', error );
+			recordNameTracks( 'stop', {
+				reason: 'error',
+				error: ( error as { message?: string } )?.message || '',
+			} );
+			setSuggestionsState( SuggestionsState.Failed );
+		},
+		onCompletionFinished: ( reason, content ) => {
```

Comment:
I did not expect that the product titles would use the streaming endpoint but it works because of this `onCompletionFinished()` function, nice thinking here!
I think I like the idea of streaming _all the things_ which then adds the possibility of rendering title suggestions one at a time as they come in, further reducing the perception of latency.
We would have to build a function that could parse the JSON string being returned from the EventSource and pick out valid product title objects, and when we find them, add them via `setSuggestions()`. 
Do you think it's worth adding as an enhancement for later?

Reply:
Definitely think this could be a worth enhancement 👍